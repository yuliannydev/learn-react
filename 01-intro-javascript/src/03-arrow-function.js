function Saludar (name) {
    return `Hola ${name}`;
}

const saludar2 = function (name) {
    return `Hola ${name}`;
}

const saludar3 = () => `holi`

const saludar4 = () => {
    return 'Holi 2'
}

console.log(Saludar('Yulianny'));
console.log(saludar2('Nazaret'));
console.log(saludar3());
console.log(saludar4());

//** return implicito */

const users = () => ({
    name: 'Yulianny',
    lastName: 'Betancourt',
})

console.log(users());