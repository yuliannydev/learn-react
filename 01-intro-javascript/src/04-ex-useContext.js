/* const getUsuarioActivo = (name) => ({
    uid: 'ABC123',
    username: name,
});

const usuarioActivo = getUsuarioActivo('Yulianny');

console.log(usuarioActivo);

const {uid, username} = usuarioActivo;
console.log(uid);
console.log(username); */

    const person = {
        uid: 'ABC123',
        username: 'Yulianny',
        lastName: 'Betancourt',
        other: {
            hobbie: 'Ver one piece',
            episode: 119,
        }
    }
    const loli = ({uid, username, city = 'Medellin', other: {hobbie, episode}}) => {
        
        return {
            id: uid,
            namae: username,
            city,
            hobbie,
            episode,
        }
    }
  
      const {id, namae, city, episode, hobbie} = loli(person);
      
      console.log(id);
      console.log(namae);
      console.log(city);
      console.log(hobbie);
      console.log(episode);