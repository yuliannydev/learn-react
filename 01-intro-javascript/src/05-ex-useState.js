/** Ejemplo para entender useState */
const onePiece = ["Luffy", "Soro", "Sanji"];
console.log(onePiece);

const [p1] = onePiece;
console.log(p1);

/** Obteniendo el segundo elemento */
const [, p2] = onePiece;
console.log(p2);

/** Obteniendo el tercer elemento */
const [, , p3] = onePiece;
console.log(p3);

/** Guardar todos los elementos excepto el primero en una variable */
const [, ...all] = onePiece;
console.log(all);

//** Desestructuracion de arreglos */

const data = () => {
  return ["ABC", 123];
};

const [letras, numeros] = data();
console.log(letras);
console.log(numeros);

//** Desestructuracion en un ejemplo de useState */

const UseState = (nombre) => {
  return [
    nombre,
    () => {
      console.log(`Hello world!, ${nombre}`);
    },
  ];
};

const [nombre, setNombre] = UseState("Nombre");
console.log(nombre);
setNombre();
