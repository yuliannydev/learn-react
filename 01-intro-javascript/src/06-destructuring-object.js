//** Destructurando objetos */
const person = {
    name : 'Yulianny',
    lastName : 'Betancourt',
    direction: {
        city: 'Medellin',
        zipcode: 50001,
    }
};

console.log(person);

let person2 = {...person};
person2.name = 'Nazaret';

console.log(person2);