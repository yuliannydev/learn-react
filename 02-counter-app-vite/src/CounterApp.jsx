import PropTypes from 'prop-types'
import { useState } from 'react'

export const CounterApp = ({value}) => {
const [ counter, setCounter ] = useState(value)

  const handleAdd = () => {
    setCounter((counter) => counter + 1)
  }

  const handleSubstract = () => {
    setCounter((counter) => counter - 1)
  }

  const handleReset = () => {
    setCounter(value)
  }

  return (
    <main>
        <h1>Counter App</h1>
        <h2> {counter} </h2>
        <button className='btn-1' onClick={() => handleAdd() }>
          +1
        </button>
        <button className='btn-2' onClick={() => handleSubstract() }>
          -1
        </button>
        <button className='btn-3' onClick={() => handleReset() }>
          Reset
        </button>

    </main>
  )
}

CounterApp.propTypes = {
    value: PropTypes.number.isRequired
}
