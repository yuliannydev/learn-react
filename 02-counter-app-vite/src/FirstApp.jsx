import React from 'react';
import PropTypes from 'prop-types';
export default function FirstApp({title, subTitle, name}) {
  return (
    <>
    <h1> {title} </h1>
    <p> {subTitle} </p>
    <em> {name} </em>
    </>
  )
}

FirstApp.propTypes = {
  title: PropTypes.string.isRequired,
  subTitle: PropTypes.string,
}

FirstApp.defaultProps = {
  name: 'Nathalie Rivas',
  title: 'Hola poshito',
  subTitle: 'Desde tu mente :3'
}