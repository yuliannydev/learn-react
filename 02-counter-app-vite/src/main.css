html, body {
    background-color: #21232a;
    color: white;
    font-family: Georgia, 'Times New Roman', Times, serif;
    font-size: 1.3rem;
    padding: 70px;
}

.btn-1, .btn-2, .btn-3 {
    padding: 0.6rem;
    border-radius: 0.5rem;
    font-weight: bold;
    font-size: 1.1rem;
    margin-right: 0.6rem;
}
.btn-1 {
    color: rgb(43, 190, 43);
}
.btn-2 {
    color: red;
}
.btn-3 {
    color: rgb(108, 101, 101);
}