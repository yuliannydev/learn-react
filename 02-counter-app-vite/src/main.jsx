import React from 'react';
import ReactDOM from 'react-dom/client';
import './main.css'
// import { CounterApp } from './CounterApp';
import FirstApp from './FirstApp';

ReactDOM.createRoot( document.getElementById('root') ).render(
    <React.StrictMode>
        {/* <CounterApp value={0}/> */}
      <FirstApp title = 'Hello world!' subTitle = 'Desde React' name = 'Yulianny Betancourt'/>
    </React.StrictMode>
)