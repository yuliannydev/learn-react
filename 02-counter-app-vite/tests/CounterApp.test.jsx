import React from 'react'
import { fireEvent, getByText, render, screen } from '@testing-library/react'
import { CounterApp } from '../src/CounterApp'

describe('Pruebas en el <CounterApp />', () => { 
    const value = 100;
    
    test('Debe hacer match con el snapshot', () => { 
       const { container } = render(<CounterApp value={value}/>)
       expect(container).toMatchSnapshot();
     })

    test('Debe mostrar el valor inicial de 100 <CounterApp value={100} />', () => { 
    
        render(<CounterApp value={value}/>)
        
        expect(Number(screen.getByRole('heading', {level: 2}).innerHTML)).toBe(value);
     })

    test('Debe incrementar con el boton +1', () => { 

        render(<CounterApp value={value} />)
        fireEvent.click(screen.getByText('+1'))
        expect(screen.getByText(101)).toBeTruthy();
     })

    test('Debe decrementar con el boton -1', () => { 
        
        render(<CounterApp value={value} />)
        /** Disparar un evento - Pruebas por texto */
        fireEvent.click(screen.getByText('-1'))
        //screen.debug();
        expect(screen.getByText(99)).toBeTruthy();
     })

    test('Debe funcionar el boton de reset', () => { 
        
        render(<CounterApp value={value} />)
        //fireEvent.click(screen.getByText('Reset'))
        //screen.debug();
        /** Pruebas por rol con aria-label */
        fireEvent.click(screen.getByRole('button', {name: 'reset'}))
        expect(screen.getByText(value)).toBeTruthy();
    })
 })