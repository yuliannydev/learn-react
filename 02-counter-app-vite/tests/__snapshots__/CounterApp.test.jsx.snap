// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en el <CounterApp /> Debe hacer match con el snapshot 1`] = `
<div>
  <main>
    <h1>
      Counter App
    </h1>
    <h2>
      100
    </h2>
    <button
      class="btn-1"
    >
      +1
    </button>
    <button
      class="btn-2"
    >
      -1
    </button>
    <button
      aria-label="reset"
      class="btn-3"
    >
      Reset
    </button>
  </main>
</div>
`;
